core/src/components/datetime/datetime.tsx

```diff
@@ -1259,6 +1233,66 @@ export class Datetime implements ComponentInterface {
        */
       this.activeParts = [];
     }
+
+    /**
+     * Only animate if:
+     * 1. We're using grid style (wheel style pickers should just jump to new value)
+     * 2. The month and/or year actually changed (otherwise there's nothing to animate to)
+     * 3. The calendar body is visible (prevents animation when in collapsed datetime-button, for example)
+     * 4. The month/year picker is not open (since you wouldn't see the animation anyway)
+     */
+    const didChangeMonth = month !== workingParts.month || year !== workingParts.year;
+    const bodyIsVisible = el.classList.contains('datetime-ready');
+    const { isGridStyle, showMonthAndYear } = this;
+    if (isGridStyle && didChangeMonth && bodyIsVisible && !showMonthAndYear) {
```

Comment:
If you set datetime to a time it still scrolls to another month and then abruptly resets to the selected month.

Example:

```html
<ion-datetime value="2022-02-22T16:30:00"></ion-datetime>

<script>
setTimeout(() => {
  datetime.value = '16:30';
}, 2000);
</script>
```

Reply:
I fixed the wonky scrolling in https://github.com/ionic-team/ionic-framework/pull/27806/commits/fbf2f9c4c526fb0b6047e977309bbba7b2440952. The datetime now errors out if you do this, but it looks like this case (setting the value to just a time when the `presentation` is something other than just `time`) was never supported; I'm able to trigger the same error on `main` after messing with it a bit. (Screencast below.) I'm thinking it would be good to add a warning for this improper usage in a separate ticket. Thoughts?

https://github.com/ionic-team/ionic-framework/assets/90629384/0bac133b-0804-4574-aee3-0f9666535e25

Reply:
Ticket created: https://ionic-cloud.atlassian.net/browse/FW-4996